### PR TITLE
Update client's configuration methods with stable config api 

### PR DESCRIPTION
**Pull request by hunter007**
# Description

_Please explain the changes you've made_

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: #556 

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [ ] Created/updated tests
* [ ] Extended the documentation

### CODE REVIEW

**Comment by shivamkm07** [comment]
Subscribe and Unsubscribe method for sdk needs to be updated as per sdk-spec(https://github.com/dapr/sig-sdk-spec/blob/main/spec/alpha/configuration.md). It's also discussed here(https://github.com/dapr/python-sdk/issues/488#issuecomment-1427037534). Subscribe/unsubscribe must happen with subscription Id and not keys. I am creating a PR for the same. This PR shouldn't be merged before that.

**Comment by shivamkm07** [comment]
> Subscribe and Unsubscribe method for sdk needs to be updated as per sdk-spec(https://github.com/dapr/sig-sdk-spec/blob/main/spec/alpha/configuration.md). It's also discussed here([#488 (comment)](https://github.com/dapr/python-sdk/issues/488#issuecomment-1427037534)). Subscribe/unsubscribe must happen with subscription Id and not keys. I am creating a PR for the same. This PR shouldn't be merged before that.

I have created PR https://github.com/dapr/python-sdk/pull/558 with the required change. Only after it is merged, the current PR should be merged. cc @berndverst

File: dapr/aio/clients/grpc/client.py
Comment (berndverst) [inline dapr/aio/clients/grpc/client.py:150]: Thanks :)

**Comment by berndverst** [comment]
@hunter007 just to double check - did you make sure to use `grpcio-tools<1.49` to generate the protos? We must use an older version because there was a breaking change in `protobuf` and we must stay on the older `protobuf` version due to compatibility with many ML SDKs.

**Comment by berndverst** [comment]
@hunter007 we made some changes to the configuration API -- could you please pull the latest changes from the master branch and resolve conflicts, then update this PR?

I'll merge your PR once that is done! Please keep in mind to generate the protos using grpcio-tools 1.48.2 (< 1.49)

Thanks

**Comment by hunter007** [comment]
> @hunter007 just to double check - did you make sure to use `grpcio-tools<1.49` to generate the protos? We must use an older version because there was a breaking change in `protobuf` and we must stay on the older `protobuf` version due to compatibility with many ML SDKs.

In my environment:

grpcio-tools==1.47.0

Is it OK?

File: dapr/aio/clients/grpc/client.py
Comment (hunter007) [reply dapr/aio/clients/grpc/client.py]: object has no __adel__ method, and self._channel also has no __del__ ,so I think no need for __del__

File: dapr/aio/clients/grpc/client.py
Comment (shivamkm07) [inline dapr/aio/clients/grpc/client.py:968]:
```suggestion
                    keys=['key_1'],
```

File: dapr/aio/clients/grpc/client.py
Comment (shivamkm07) [inline dapr/aio/clients/grpc/client.py:966]:
```suggestion
                resp = await d.subscribe_configuration(
```

File: dapr/clients/grpc/client.py
Comment (shivamkm07) [inline dapr/clients/grpc/client.py:955]:
```suggestion
                    keys=['key_1'],
```

File: dapr/clients/grpc/client.py
Comment (shivamkm07) [inline dapr/clients/grpc/client.py:953]:
```suggestion
                resp = d.subscribe_configuration(
```

File: dapr/aio/clients/grpc/client.py
Comment (shivamkm07) [inline dapr/aio/clients/grpc/client.py:1004]:
```suggestion
        response: UnsubscribeConfigurationResponse = await self._stub.UnsubscribeConfiguration(req)
```

**Comment by berndverst** [comment]
> > @hunter007 just to double check - did you make sure to use `grpcio-tools<1.49` to generate the protos? We must use an older version because there was a breaking change in `protobuf` and we must stay on the older `protobuf` version due to compatibility with many ML SDKs.
> 
> In my environment:
> 
> ```
> ➜  python-sdk git:(master) pip3 freeze|grep grpcio
> grpcio==1.47.0
> grpcio-tools==1.47.0
> ➜  python-sdk git:(master) protoc --version
> libprotoc 3.21.1
> ```
> 
> Is it OK?

Yes that's ok! Thanks for confirming.

**Review by berndverst — APPROVED** [review]

**Pull request merged**
